apps/codeforafrica/src/components/Opportunities/Opportunities.js

```diff
@@ -41,22 +46,24 @@ const Opportunities = React.forwardRef(function Opportunities(
 
   const handleChangeTag = (_, value) => {
     const newValue =
-      (value && tags.find((t) => equalsIgnoreCase(value, t))) || ALL_TAG;
+      (value && tags.find((t) => equalsIgnoreCase(value, t.slug))) || allTag;
     setTag(newValue);
     setPage(1);
   };
 
-  const { data } = useOpportunities({ page, q, tag });
+  const { data } = useOpportunities({ page, q, tag: tag.slug }, primaryTag);
   useEffect(() => {
     if (data) {
-      const { results, pagination } = data;
+      const { posts: results, pagination } = data;
       setCount(pagination.count);
       setOpportunities([...results]);
     }
   }, [data]);
 
   useEffect(() => {
-    router.push(queryParams, undefined, {
+    const { pathname } = window.location;
+    const url = pathname ? `${pathname}${queryParams}` : queryParams;
```

Comment:
```suggestion
    const url = `${pathname}${queryParams}`;
```

`pathname` should always be there, no?